app/components/paywall_component.rb

```diff
@@ -0,0 +1,22 @@
+class PaywallComponent < ApplicationComponent
+  def initialize(user:, paywalled:)
+    @user = user
+    @paywalled = paywalled
+  end
+
+  def render?
+    customer? || owner?
+  end
+
+  private
+
+  def customer?
+    @user.active_business_subscription?
+  end
+
+  def owner?
+    @paywalled.user == @user
+  rescue
```

Comment:
Heads up that if we do a nil-coalescing operator then we will need a check for when both are `nil` (I think).

Reply:
Switched to the safe navigation operator for this. I don't think we need to check for double nil though. I added a test in to check for user being `nil`, and everything passes as we'd expect.

Reply:
I don't know why I brainfarted on not using the safe navigation operator though. Was a super easy use case here.

Reply:
**Update:**

Sorry for blowing this thread up. I realized why safe navigator doesn't work here. It's not just about catching the paywalled content being `nil`. You also have to account for bad inputs, which requires a `rescue` clause.

Reply:
Ideally, you never `rescue` without specifying a specific error or errors to catch. Otherwise it can "catch" tons of stuff, like typos! For example,

```ruby
[1] pry(main)> def foo
[1] pry(main)*   asdjaksdjk
[1] pry(main)* rescue
[1] pry(main)*   "rescued!"
[1] pry(main)* end
=> :foo
[2] pry(main)> foo
=> "rescued!"
```

So even if we don't use the `&`, let's explicitly check for what we expect. I think here that means seeing if `@paywalled` responds to `#user`, right? That could be something like the following.

```ruby
def owner?
  if @paywalled.respond_to?(:user) && @paywalled == @user
    true
  else
    false
  end
```

But now that I write that, what are the scenarios where `@paywalled` _doesn't_ respond to `#user`? Is this overly defensive programming or an actual use-case that popped up in the PR? I'm also wondering, if it doesn't respond, why would it be false and not true?

Reply:
Oh, you're totally right. That was dumb of me with the rescue block. I'm of the opinion it was overly defensive programming on my part. I'll revise this after work today if I have time.

Getting some cavities filled and then seeing a movie with my girlfriend, so I'm not sure if I'll get around to it.

Reply:
Enjoy! Well, the movie, not the cavities. And no rush, whenever you get around to it is fine by me.